Задание по спринту №18

CasTeTRU opened the pull request.

CasTeTRU committed:
Задание по спринту №18

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/library/ui/LibraryState.kt:
```diff
@@ -1,5 +1,5 @@
 
 data class LibraryState(
-    val imageUrl: String = "https://img.freepik.com/free-vector/open-blue-book-white_1308-69339.jpg"
+    val currentTab: Int = 0
```
TabLayout в Android автоматически сохраняет текущий выбор, поэтому хранить отдельно ещё в стейте не стоит - так как по-хорошему должен быть только 1 источник правды

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/library/ui/PlaylistsFragment.kt:
```diff
@@ -0,0 +1,48 @@
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_playlists, container, false)
```
Рекомендую сразу воспользоваться `viewBinding` во фрагментах. Почитать о том, как использовать эту фичу во фрагментах можно тут - https://developer.android.com/topic/libraries/view-binding#fragments.

ArturNurtdinov reviewed (commented):
Выделение активного элемента TabLayout не соответствует макетам в светлой теме. Также табы перепутаны местами - сначала должны быть "Избранные треки", а потом - "Плейлисты":

<img width="855" height="477" alt="image" src="https://github.com/user-attachments/assets/a8a9bb66-e60d-4ceb-924d-27c4e868734c" />

Макет:

<img width="1181" height="587" alt="image" src="https://github.com/user-attachments/assets/9f94eddc-80e1-494b-8553-5d78a7d34674" />

CasTeTRU committed:
Задание по спринту №18
1) Поправлен порядок вкладок (Избранное/плейлисты). задание цвета индикатора активной вкладки через tabIndicatorColor для светлой темы вообще никак не реагирует на задание цвета, работает только для тёмной
2) Убран LibraryState
3) Использован viewBinding для фрагментов

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/library/ui/MediatekaActivity.kt:
```diff
@@ -30,25 +24,21 @@ class MediatekaActivity : AppCompatActivity() {
             tab.text = when (position) {
-                0 -> getString(R.string.playlists)
-                1 -> getString(R.string.favorite_tracks)
+                0 -> getString(com.example.playlistmaker.R.string.favorite_tracks)
+                1 -> getString(com.example.playlistmaker.R.string.playlists)
```
Здесь лучше сделать импорт к R классу, чтобы в коде не указывать весь путь к классу
